GameServer/gameutils/PlayerTradeWindow.cs

```diff
@@ -648,13 +659,24 @@ public bool AcceptTrade()
 
 						bool tradeSuccess = false;
 
+                        InventoryItem itemtoadd = item;
+
+                        // If PLayer is not Infiltrator (9), Shadowblade (23), Nightshade (49), remove Envenom bonus before add the item in the inventory
+                        if (item.PoisonSpellID > 0 && !(m_owner.CharacterClass.ID == 9 || m_owner.CharacterClass.ID == 23 || m_owner.CharacterClass.ID == 49))

```

Comment:
The comment from @NetDwarf applies here too I guess, let's just remove envenom bonus from everyone.